File: root/common/src/main/java/org/sc/common/rest/controller/FileDownloadResponse.java
package org.sc.common.rest.controller;

import java.util.Collections;
import java.util.Set;

public class FileDownloadResponse extends RESTResponse {
    private final String path;

    public FileDownloadResponse(String path) {
        super(Status.OK, Collections.emptySet());
        this.path = path;
    }

    public FileDownloadResponse(String path, Status error, Set<String> errors) {
        super(error, errors);
        this.path = path;
    }

    public String getPath() {
        return path;
    }
}

File: root/rest/src/main/java/org/sc/common/rest/response/FileDownloadResponse.kt
package org.sc.common.rest.response

import org.sc.common.rest.Status

data class FileDownloadResponse(
    val status: Status,
    val message: Set<String>,
    val path: String,
    override val currentPage: Long,
    override val totalPages: Long,
    override val size: Long,
    override val totalCount: Long
) :
    RESTResponse(currentPage, totalPages, size, totalCount)